src/Algebra/Properties/Monoid/Reasoning.agda

```diff
@@ -0,0 +1,70 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Equational reasoning for monoids
+-- (Utilities for identity and cancellation reasoning, extending semigroup reasoning)
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Algebra.Bundles using (Monoid)
+
+module Algebra.Properties.Monoid.Reasoning {o ℓ} (M : Monoid o ℓ) where
```

Comment:
As usual: these lemmas belong in `Algebra.Properties.Monoid`, not in a separate module as you have it.

By all means do initial development in that style before raising a PR, or else mark it as `DRAFT` (and hence not ready-for-review) but once you do raise the PR/ask for review, we do need it to be compatible both with the existing `stdlib` hierarchy/structure, but also with the `style-guide`, or else we will expend a lot of extra reviewing energy on things which should have been fixed ahead of time.